pypeit/core/moment.py

```diff
@@ -0,0 +1,497 @@
+                                [(x_i - \mu_1)^2 + \mu_2]^2.
+
+    The values returned for the second-moment calculation are actually
+    the standard deviation instead of the variance, where:
+
+    .. math::
+        \sigma = \sqrt{\mu_2} \\
+        \epsilon_{\sigma} = \frac{\epsilon_{\mu_2}}{2 \sigma}.
+
+    The method uses `numpy.ma.MaskedArray` objects to keep track of
+    math errors, such as divisions by 0. The returned boolean array
+    indicates when these errors occurred, and the method replaces
+    these errors with the original centers.
+
+    The shape of arrays depend on the shapes of the input image
+    (`flux`), columns about which to perform the calculation (`col`),
+    and rows from which to select the column data (`row`) as listed
+    below.  In the following, the input flux array must be 2D and has
+    shape :math:`(N_{\rm row}, N_{\rm col})`, whereas the number of rows
+    and columns identified for a moment calculation are :math:`N_{\rm
+    mom,row}` and :math:`N_{\rm mom,col}`, respectively.  The possible
+    shapes of the output arrays are:
+
+        - If `row` is None and `col` is 1D, the moments are calculated
+          centered at each `col` for all rows in `flux`. The shape of
+          the array for each moment is then :math:`(N_{\rm row}, N_{\rm
+          mom,col})`
+        - If `row` is None and `col` is 2D, `col` must have shape
+          :math:`(N_{\rm row}, N_{\rm mom,col})` and the shape of the
+          output array per moment calculated is the same.
+        - If `row` is a single integer and `col` is 1D, the moments are
+          calculated for each col only at the provided row in `flux`.
+          The shape of the array for each moment is then :math:`(N_{\rm
+          mom,col},)`.
+        - If `row` is a 1D integer array and `col` is 1D, the default
+          behavior is for the moments to be calculated at each col
+          for each provided row. The shape of the array for each
+          moment would then be :math:`(N_{\rm mom,row},N_{\rm
+          mom,col})`. This will be true even if :math:`N_{\rm
+          mom,row} = N_{\rm mom,col}`, as long as `mesh` is `True`.
+          However, if :math:`N_{\rm mom,row} = N_{\rm mom,col}` and
+          `mesh` is `False`, only the matched pairs of `col` and
+          `row` will be used and the output shape per moment will be
+          :math:`(N_{\rm mom,row},)`.
+        - If `row` is a 1D integer array and `col` is 2D, the first axis
+          of `col` must be the same length as `row`. The shape of the
+          array for each moment is then :math:`(N_{\rm mom,row},N_{\rm
+          mom,col})`, which is the same as the input `col`.
+        - If `row` is 2D and `col` is 2D, they must have the same shape.
+        - If `row` is a single integer and `col` is 2D, or if `row` is
+          2D and `col` is 1D, the method raises an error.
+
+    .. warning::
+
+        The function has significant setup/input checking overhead.
```

Comment:
Here's the new comment:
```
        The function has significant setup/input checking. Most of
        this introduces limited overhead with the exception of the
        handling of `ivar`, `bpm`, and `fwgt`. If any of these are
        provided as `None` on input, an array is constructed (unity
        for `ivar` and `fwgt` and all False for `bpm`) that serves as
        a place holder. If repetitive calls to the function are
        expected and any of these arrays are missing, significant
        efficiency gains can be made by providing pre-built values
        for these arrays so that time isn't lost in allocating the
        placeholder arrays in every call.
```
And see the benchmark I put in the #trace_slits slack channel.